src/coreclr/jit/lowerxarch.cpp

```diff
@@ -4006,6 +4006,36 @@ GenTree* Lowering::TryLowerAndOpToAndNot(GenTreeOp* andNode)
     return andnNode;
 }
 
+void Lowering::LowerBswapOp(GenTreeOp* node)
+{
+    assert(node->OperIs(GT_BSWAP));
+
+    if (!comp->compOpportunisticallyDependsOn(InstructionSet_MOVBE))
```

Comment:
Probably a good idea to guard this under `comp->opts.OptimizationEnabled()` too.